### PR TITLE
Minor inline documentation & README improvements

### PR DESCRIPTION
**Pull request by cdzombak**
While reviewing the code, I found a few places where the inline documentation could be clearer; and I also wanted to note the getting-started steps with Carthage in case other devs who check out this repo aren't already familiar.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -227,6 +227,10 @@ pod 'YapDatabaseExtensions/Functional’
 
 API documentation is available on [CocoaDocs.org](http://cocoadocs.org/docsets/YapDatabaseExtensions).
 
+## Developing
+
+To start working in this repository’s `YapDatabaseExtensions.xcodeproj`, you’ll need to use [Carthage](https://github.com/Carthage/Carthage#installing-carthage) to download & build the project’s dependencies, with the commands `carthage checkout` and `carthage build`.
```
Comment (cdzombak) [inline README.md:232]: Hm, [this helpful blog post from Thoughtbots](https://robots.thoughtbot.com/creating-your-first-ios-framework) indicates that the following command should be used:

```
carthage update --platform iOS --use-submodules --no-use-binaries
```

@danthorpe I haven't used Carthage much before. How have you been achieving this on your machine?

**Comment by danthorpe** [comment]
@cdzombak I've been okay with `$ carthage update`. Sometimes using `--no-use-binaries` to force a local build. Are you having a problem with it?

I've not done any development since before Xcode 7.2 - there might be a problem, as some of the Persistable API no longer works. It's on my list to fix shortly.

**Comment by cdzombak** [comment]
> I've been okay with `$ carthage update`. … Are you having a problem with it?

No, I haven't faced any problem with it, but I am wondering about best practice and what's been working for you (again, I'm relatively new to Carthage).

My first instinct when I checked out the repo was to run `carthage update`, but that resulted in a change to [`Cartfile.resolved`](https://github.com/danthorpe/YapDatabaseExtensions/blob/development/Cartfile.resolved#L3). The `checkout`/`build` approach seemed to build the right dependencies without touching that file, and that seemed like the "right" behavior for someone who's just checked out the repo.

**Comment by danthorpe** [comment]
Arr yes, to build exactly the same dependencies that sounds about right. I don't think there is any need to worry about the changes to `Carthfile.resolved` - there was a new version of ValueCoding to support nested sequence types.

**Pull request merged**